sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -292,10 +292,12 @@ private[hive] class HiveClientImpl(
   }
 
   override def setCurrentDatabase(databaseName: String): Unit = withHiveState {
-    if (databaseExists(databaseName)) {
-      state.setCurrentDatabase(databaseName)
-    } else {
-      throw new NoSuchDatabaseException(databaseName)
+    if (state.getCurrentDatabase != databaseName) {
+      if (databaseExists(databaseName)) {
```

Comment:
This PR uses an additional `getCurrentDatabase` to avoid `databaseExists`. Can we have a more specific in the title?